### ISSUE TITLE
`ImageSource.images()` can only load sequential images

### ISSUE THREAD

**Issue by chris-langfield** [ComputationalCryoEM/ASPIRE-Python#524]
`ImageSource.images()` takes mandatory `start` and `num` parameters.

However, after `start` and `num` are specified, an `indices` array is created via:
```python
 indices = np.arange(start, min(start + num, self.n), dtype=int)
```

This is then passed as a kwarg to the subclass `_images()`. 

This makes it impossible to ask for specific, nonsequential indices when calling the public method:

```python
>>> src = RelionSource("sample_relion_data.star")
>>> src.images(indices=[7,2,14,10])
TypeError: images() missing 2 required positional arguments: 'start' and 'num'
>>> src.images(0,17, indices=[7,2,14,10])
TypeError: _images() got multiple values for keyword argument 'indices'
```

The private `_images` methods of `ImageSource` subclasses, including `RelionSource`, are written in a way that allows for the `indices` to be nonsequential. The only way to ask for indices in this way is to call the private method, but we would like the public method to be able to do this.

**Comment by janden** [comment]
So the reason for the (public) `(start, num)` interface is that we want to highlight that the natural way to get images is by sequential batches. Often this will result in faster access and more efficient processing. However, as you point out, this is not always desired. The probability that someone will “accidentally” as for non-sequential images is quite low.

It's not clear to me why the `indices` argument was added (it was not in the original Matlab code), but we could make that the default interface. The important thing here would be to make sure that we don't lose any of the efficiencies when `indices` is a range.

**Comment by garrettwrong** [comment]
An obvious place we want to use indices is after classification.  The thing is if we roll our own loop there, then roll it somewhere else for another problem, we might have well just centralized it into the sources...

**Comment by janden** [comment]
Of course, I would lean to switching this to accepting a set of indices, again as long as we're able to detect when sequential loading is possible and take advantage of this. I think we currently are able to do this with the `groupby` approach in `_images`.

**Comment by chris-langfield** [comment]
Closed by #681
